lib/rubocop/cop/rspec/unexpected_requires.rb

```diff
@@ -0,0 +1,60 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module RSpec
+      # Checks that rspec files do not contain requires, as this
+      # can lead to unexpected behavior later when the code is
+      # not used with rspec.
+      #
+      # Require the necessary files in the projects config or
+      # where you need it instead.
+      #
+      # @example
+      #   # bad
+      #   require "lib/use_cases/my_work"
+      #
+      #   describe UseCases::MyWork do
+      #     ...
+      #   end
+      #
+      #   # good
+      #   describe UseCases::MyWork do
+      #     ...
+      #   end
+      #
+      class UnexpectedRequires < Base
+        MSG = 'Do not require anything in a test file.'
+
+        IGNORED_PATTERN = /(spec|rails)_helper/
+
+        def on_new_investigation
+          return if processed_source.path.match?(IGNORED_PATTERN)
+
+          unnecessary_requires.each do |token|
+            add_offense(token.pos, message: MSG)
+          end
+        end
```

Comment:
I think the whole implementation can be a lot simpler, e.g.
```rb
        def on_new_investigation
          @ignore_file = processed_source.path.match?(IGNORED_PATTERN)
        end

        def on_send(node)
          return if @ignore_file

          foo_node_matcher { add_offense(node) }
        end
```
(and then define the actual node matcher).

But we need to address @pirj’s question about legitimate requires first.

Reply:
require 'spec_helper' and require 'rails_helper' should be allowed, so a simple node matcher for require is not good enough.

Would then look like this. 
I think adding the exclusion of spec/rails helper to the regex would make it a) hard to write b) hard to read c) hard to understand ... 


```
        def_node_matcher :require_symbol?, <<~PATTERN
          (send #rspec? :require $_ ...)
        PATTERN
        
                def on_new_investigation
          @ignore_file = processed_source.path.match?(IGNORED_PATTERN)
        end

        def on_send(node)
          return if @ignore_file

          require_symbol?(node) do |match|
            next if match.value.match?(IGNORED_PATTERN)

            add_offense(node, message: MSG)
          end
        end
```